server/service/osquery.go

```diff
@@ -513,6 +513,19 @@ func (svc *Service) GetDistributedQueries(ctx context.Context) (queries map[stri
 		discovery[name] = query
 	}
 
+	// The following is added to improve Fleet Desktop's UX at install time.
+	//
+	// At install (enroll) time, the "orbit_info" extension takes longer to load than the first
+	// query check-in (distributed/read request).
+	// To avoid having to wait for the next check-in to ingest the data (after
+	// svc.config.Osquery.DetailUpdateInterval, 1h by default),
+	// we make the best effort to retrieve such "device auth token" from the device, but with a
+	// limit of 1m to not generate too much write database overhead (writes to `host_device_auth` table).
+	if time.Now().UTC().Sub(host.LastEnrolledAt) < 1*time.Minute {
```

Comment:
just for my own understanding: I assume this is an heuristic value, but it hits me as too low of a threshold, as my understanding is that the default value of `distributed_interval` in osquery is 1 minute. What was your process for finding this value? can we increment the time limit but maybe also check if the `orbit_info` is already there or not?

Reply:
This is assuming the default value for distributed_interval set by Fleet to `10s`.

![Screen Shot 2022-05-26 at 16 53 11](https://user-images.githubusercontent.com/2073526/170566736-fe404c1f-da6a-4493-a281-b12096491d81.png)

> can we increment the time limit but maybe also check if the orbit_info is already there or not?

Indeed. I thought of this, am all ears, would loading from the row from `host_device_auth` and only write if "the token is not set or has changed" make this have less overhead?

Reply:
hm, I naively assumed we could get relevant information to do the check from the host. I don't feel I have a good grasp to provide useful input on that, so I'd trust your gut (I was mostly interested in understanding your thought process)

there's the edge case of a config with a larger `distributed_interval`, but I'd say let's give it a go as-is and see what happens.